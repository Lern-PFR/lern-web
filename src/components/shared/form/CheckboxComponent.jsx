import styled from 'styled-components';
import {
	border,
	color,
	display,
	flexbox,
	layout,
	margin,
	padding,
	position,
} from 'styled-system';
import PropTypes from 'prop-types';
import { disabled as disabledStyle } from 'theme/buttonStyles';

const StyledCheckboxComponent = styled('input')(
	{
		cursor: 'pointer',
	},
	border,
	color,
	display,
	flexbox,
	layout,
	margin,
	padding,
	position,
);

/**
 * @name CheckboxComponent
 * @description A component used to display styled checkbox element.
 *
 * @param {bool}	[disabled]		: Whether the checkbox is disabled.
 * @param {bool}	[checked]		: Whether the checkbox is checked.
 * @param {string}	id				: The id of the checkbox.
 */
const CheckboxComponent = ({ id, checked, disabled, ...otherProps }) => (
	<StyledCheckboxComponent {...otherProps} {...(disabled ? disabledStyle : {})} type="checkbox" disabled={disabled} defaultChecked={checked} id={id} />
);

CheckboxComponent.propTypes = {
	id: PropTypes.string.isRequired,
	disabled: PropTypes.bool,
	checked: PropTypes.bool,
};

CheckboxComponent.defaultProps = {
	disabled: false,
	checked: false,
};

export default CheckboxComponent;
