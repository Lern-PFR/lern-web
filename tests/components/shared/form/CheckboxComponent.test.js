import { shallow } from 'enzyme';
import CheckboxComponent from 'components/shared/form/CheckboxComponent';
import { cleanup } from '@testing-library/react';

describe('CheckboxComponent', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match previous snapshot', () => {
		const sut = (<CheckboxComponent id="cb" />);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot for checked state', () => {
		const sut = (<CheckboxComponent checked id="cb" />);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot for disabled state', () => {
		const sut = (<CheckboxComponent disabled id="cb" />);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match previous snapshot for disabled and checked states', () => {
		const sut = (<CheckboxComponent disabled checked id="cb" />);
		const wrapper = shallow(sut);

		expect(wrapper).toMatchSnapshot();
	});
});
