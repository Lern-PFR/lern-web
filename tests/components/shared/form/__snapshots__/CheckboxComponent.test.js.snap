// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxComponent should match previous snapshot 1`] = `
<styled.input
  defaultChecked={false}
  disabled={false}
  id="cb"
  type="checkbox"
/>
`;

exports[`CheckboxComponent should match previous snapshot for checked state 1`] = `
<styled.input
  defaultChecked={true}
  disabled={false}
  id="cb"
  type="checkbox"
/>
`;

exports[`CheckboxComponent should match previous snapshot for disabled and checked states 1`] = `
<styled.input
  bg="F6F6F4"
  color="#8C8C8C"
  defaultChecked={true}
  disabled={true}
  id="cb"
  type="checkbox"
/>
`;

exports[`CheckboxComponent should match previous snapshot for disabled state 1`] = `
<styled.input
  bg="F6F6F4"
  color="#8C8C8C"
  defaultChecked={false}
  disabled={true}
  id="cb"
  type="checkbox"
/>
`;
